libbeat/common/mapstr.go

```diff
@@ -101,12 +101,10 @@ func (m MapStr) Clone() MapStr {
 	result := MapStr{}
 
 	for k, v := range m {
-		innerMap, err := toMapStr(v)
-		if err == nil {
-			result[k] = innerMap.Clone()
-		} else {
-			result[k] = v
+		if innerMap, ok := tryToMapStr(v); ok {
```

Comment:
ups, kind of leaked into this PR :)
I found the old processors ordering potentially overwriting a globally shared MapStr and was forced to introduce some `Clone()`, so the processors always operate on copies of globally shared MapStr instances. The amount of `fmt.Errorf` was so expensive it did totally kill throughput.